flutter/lib/src/sentry_flutter_options.dart

```diff
@@ -183,22 +181,35 @@ class SentryFlutterOptions extends SentryOptions {
 
   /// Automatically attaches a screenshot when capturing an error or exception.
   ///
-  /// Requires adding the [SentryScreenshotWidget] to the widget tree.
+  /// Requires adding the [SentryWidget] to the widget tree.
   /// Example:
-  /// runApp(SentryScreenshotWidget(child: App()));
-  /// The [SentryScreenshotWidget] has to be the root widget of the app.
-  bool attachScreenshot = false;
+  /// runApp(SentryWidget(child: App()));
+  /// The [SentryWidget] has to be the root widget of the app.
+  @Deprecated('Use `screenshot.attach` instead')
+  bool get attachScreenshot => screenshot.attach;
+  set attachScreenshot(bool value) => screenshot.attach = value;
 
   /// The quality of the attached screenshot
-  SentryScreenshotQuality screenshotQuality = SentryScreenshotQuality.high;
+  @Deprecated('Use `screenshot.quality` instead')
+  SentryScreenshotQuality get screenshotQuality => screenshot.quality;
+  set screenshotQuality(SentryScreenshotQuality value) =>
+      screenshot.quality = value;
 
   /// Only attach a screenshot when the app is resumed.
-  bool attachScreenshotOnlyWhenResumed = false;
+  @Deprecated('Use `screenshot.attachOnlyWhenResumed` instead')
+  bool get attachScreenshotOnlyWhenResumed => screenshot.attachOnlyWhenResumed;
+  set attachScreenshotOnlyWhenResumed(bool value) =>
+      screenshot.attachOnlyWhenResumed = value;
 
   /// Sets a callback which is executed before capturing screenshots. Only
   /// relevant if `attachScreenshot` is set to true. When false is returned
   /// from the function, no screenshot will be attached.
-  BeforeScreenshotCallback? beforeScreenshot;
+  @Deprecated('Use `screenshot.beforeCapture` instead')
+  BeforeScreenshotCallback? get beforeScreenshot => screenshot.beforeCapture;
```

Comment:
I'm not sure whether we should keep this consistent with other sdks by keeping`before` callbacks in the root options.

We're also planning on adding a `beforeCaptureViewHierarchy` so having this within screenshot would be inconsistent then, our other sdks also don't nest this

Reply:
I understand. It makes sense to keep it consistent with the other `SDKs`. I will move the `SentryScreenshotOptions` into the `SentryOptions`.